src/App.tsx

```diff
@@ -1,26 +1,506 @@
+
+              return null;
+            } finally {
+              setIsUpdating(null);
+            }
+          }
+
+          return null;
+        });
+
+        const updatedTodo = await Promise.all(updatedTodos);
+
+        const successfulUpdated = updatedTodo.filter(
+          id => id !== null,
+        ) as number[];
+
+        setTodos(currentTodos => {
+          return currentTodos.map(todo => {
+            if (successfulUpdated.includes(todo.id)) {
+              return {
+                ...todo,
+                completed: true,
+              };
+            } else {
+              return todo;
+            }
+          });
+        });
+
+        setAllTodos(currentTodos => {
+          return currentTodos.map(todo => {
+            if (successfulUpdated.includes(todo.id)) {
+              return {
+                ...todo,
+                completed: true,
+              };
+            } else {
+              return todo;
+            }
+          });
+        });
+
+        handleFilter(sortBy);
+      };
+
+      updateTodos();
+    }
+
+    if (hasUncompleteTodo < 0) {
+      const updatedStatus = { completed: false };
+
+      const updateTodos = async () => {
+        const updatedTodos: Promise<number | null>[] = todos.map(async todo => {
+          setIsUpdating(todo.id);
+          setErrorMessage(null);
+
+          try {
+            await updateTodo(todo.id, updatedStatus);
+
+            setItemsLeft(currentItemsLeft => currentItemsLeft + 1);
+
+            return todo.id;
+          } catch {
+            setErrorMessage('Unable to update a todo');
+
+            return null;
+          } finally {
+            setIsUpdating(null);
+          }
+        });
+
+        const updatedTodo = await Promise.all(updatedTodos);
+
+        const successfulUpdated = updatedTodo.filter(
+          id => id !== null,
+        ) as number[];
+
+        setTodos(currentTodos => {
+          return currentTodos.map(todo => {
+            if (successfulUpdated.includes(todo.id)) {
+              return {
+                ...todo,
+                completed: false,
+              };
+            } else {
+              return todo;
+            }
+          });
+        });
+
+        setAllTodos(currentTodos => {
+          return currentTodos.map(todo => {
+            if (successfulUpdated.includes(todo.id)) {
+              return {
+                ...todo,
+                completed: false,
+              };
+            } else {
+              return todo;
+            }
+          });
+        });
+
+        handleFilter(sortBy);
+      };
+
+      updateTodos();
+    }
+  }
+
+  function handleSubmit(e: React.FormEvent<HTMLFormElement>) {
+    e.preventDefault();
+    if (query.trim() !== '') {
+      setNewTask(query);
+      setIsSubmiting(true);
+    } else {
+      setErrorMessage('Title should not be empty');
+    }
+  }
+
+  function handleDeleteTodo(id: number) {
```

Comment:
such functions are redundant - just use `setDeleteTodoId` everywhere instead of `handleDeleteTodo`